openwisp_monitoring/device/api/views.py

```diff
@@ -112,20 +112,34 @@ def _get_charts_data(self, charts, time, timezone):
 
     def _get_csv(self, data):
         header = ['time']
-        columns = [data['x']]
+        columns = [data.get('x')]
+        histograms = []
         for chart in data['charts']:
-            # TODO: add way to export data for histogram charts
             if chart['type'] == 'histogram':
+                histograms.append(chart)
                 continue
             for trace in chart['traces']:
                 header.append(self._get_csv_header(chart, trace))
                 columns.append(trace[1])
         rows = [header]
-        for index, element in enumerate(data['x']):
+        for index, element in enumerate(data.get('x', [])):
```

Comment:
I just found that this is a bug, if there is only one histogram chart in Monitoring Charts (i.e. no time series charts) and you try to export it, this leads to KeyError.
```python
  File "/home/hardik/openwisp/openwisp-monitoring/openwisp_monitoring/device/api/views.py", line 75, in get
    response = HttpResponse(self._get_csv(data), content_type='text/csv')
  File "/home/hardik/openwisp/openwisp-monitoring/openwisp_monitoring/device/api/views.py", line 125, in _get_csv
    for index, element in enumerate(data['x']):
KeyError: 'x'
```
This change fixes it

Reply:
Great, thanks